src/chameleon/tests/test_loader.py

```diff
@@ -41,6 +41,32 @@ def test_load_abs(self):
         result = self._load(loader, abs)
         self.assertEqual(result.filename, abs)
 
+    def test_load_egg(self):
+        import sys
+        import zipimport
+
+        import pkg_resources
+        fn = pkg_resources.resource_filename(
+            'chameleon',
+            'tests/inputs/chameleon_test_pkg-1.0-py3.7.egg')
+        zipimport.zipimporter(fn).load_module('chameleon_test_pkg')
+        try:
+            # we use auto_reload to trigger a call of mtime
+            loader = self._makeOne(auto_reload=True)
+            result = self._load(loader, 'chameleon_test_pkg:templates/test.pt')
+            self.assertIsNone(result._v_last_read)
+            output = result(content='Test Content')
+            self.assertIsNotNone(result._v_last_read)
+            old_v_last_read = result._v_last_read
+            self.assertIn("Test Title", output)
+            self.assertIn("Test Content", output)
+            # make sure the template isn't recooked
```

Comment:
When cooking, there's an assumption I think that we're also _loading_, i.e. invoking the loader's `get` method. Ideally then, instead of testing against the private attribute `_v_last_read`, we probably should mock the loader's `get` method.

Reply:
Not sure what you mean. The ``get`` method doesn't seem to be called from that test, most likely because of the custom ``_load`` on the test class.